### PR TITLE
increase work_mem from 4mb default to 32mb (WQP-1597)

### PR DESCRIPTION
**Pull request by kkehl-usgs**
See the defect for some data, but generally the default postgres work_mem of 4mb is considered too small for any serious application, and that's what we're using.

This is causing "lossy heap blocks" where the query is unable to write a 1 for each row that matches the index criteria and so starts writing a 1 for each block that contains a row that matches, and then searches each block later, causing a big degradation in performance.

The risk with increasing work_mem is that we might start hitting out of memory errors.   32mb is not considered to be a particular large value for this setting, so I thought we could start with this and then back off if it causes too much trouble.

### CODE REVIEW

**Comment by kkehl-usgs** [comment]
Also, the command is in isolation here in an .sh file, because it cannot run in a transaction.  It doesn't have to to run before a restart, but it has to at least run before a reload of the conf file.

File: liquibase/scripts/z0_restart.sh [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #!/bin/bash 
 
 # Restart postgres to make sure we can connect
+
+psql ${WQP_DATABASE_NAME} -c "alter system set work_mem='32MB'"
```
Comment (orosz-usgs) [inline liquibase/scripts/z0_restart.sh:5]: Looks good. Would this improve the performance of an 'alter table set logged' command?

**Review by orosz-usgs — APPROVED** [review]
Just the one question.

File: liquibase/scripts/z0_restart.sh (hunk 1 above)
Comment (kkehl-usgs) [reply liquibase/scripts/z0_restart.sh:5]: Possibly.   I can try today.   You can set work_mem temporarily in a db session with "set workmem='16MB'".  That value will affect your session only so is considered safe to do.

This change may cause out of memory errors, will have to see.  I wasn't able to find any specific guidelines about how to set work_mem relative to total RAM.  It seems to be dependent on the sort of workloads you have.  If you are doing OLTP, they say 4mb is fine.  If you are doing a smaller number of more complicated queries, as we are, generally the more work_mem the better until you hit memory limitations.

Comment (jkreft-usgs) [reply liquibase/scripts/z0_restart.sh:5]: on prod we are running r4.8xlarge, with 244gb of memory, so we are in pretty good shape, memory-wise

**Pull request merged**